crates/ruff_linter/resources/test/fixtures/refurb/FURB179_0.py

```diff
@@ -0,0 +1,83 @@
+from functools import reduce
+from itertools import chain
+from operator import add, concat, sub
+
+s = "123456789"
+b = b"123456789"
+t = ((1, 2, 3), (4, 5, 6), (7, 8, 9))
+
+l = [(1, 2, 3), (4, 5, 6), (7, 8, 9)]
+e = {(1, 2, 3), (4, 5, 6), (7, 8, 9)}
+d = {(1, 2, 3): 0, (4, 5, 6): 1, (7, 8, 9): 2}
+i = iter(l)
+
+
+### Errors
+
+# Safe: Immutable iterable
+chain(*s)
+chain(*b)
+chain(*t)
+reduce(add, s)
+reduce(concat, b)
+reduce(add, t, [])
+
+# Unsafe: Mutable iterable
+chain(*l)
+chain(*e)
+chain(*d)
+chain(*i)
+reduce(concat, l, [])
+reduce(add, e)
```

Comment:
This is only valid if you know e is a list of list/tuple, add a test where it's not (like it's just a list of numbers). This should be no error.

Reply:
That the last change needs to
Make it useful and there should be builtin utilities to determine is something is a list/tuple in ruff

Reply:
Empty list, empty tuple, and empty set would all be valid, right? It's better to be too conservative vs too aggressive. Surprised there isn't a nested utility for this.

Reply:
We can add a todo comment for the more complicated uses cases later.